src/main/scala/com/codacy/client/stash/client/StashClient.scala

```diff
@@ -191,6 +191,8 @@ class StashClient(apiUrl: String, authenticator: Option[Authenticator] = None) {
       .getOrElse(Right(json))
   }
 
-  private def generateUrl(endpoint: String) = s"$apiUrl$endpoint"
+  private def generateUrl(endpoint: String) = {
+    s"$apiUrl${endpoint.replace(" ", "-")}"
```

Comment:
What sorcery is this? :D

Reply:
Quick magic to deal with the fact that Bitbucket allows spaces in the slugs when creating repositories and returns it that way to us via their API, however it doesn't accept the same URL back in, it requires the space to be replaced with a `-`. Quick wins; if you've got another quick suggestion however?

Reply:
Magic! It actually isn't supposed to be in this PR!